cmake/modules/unittest.cmake

```diff
@@ -96,9 +96,11 @@ if(COVERAGE)
     -ftest-coverage
     )
 
-  target_link_libraries(testbinary PRIVATE
-    -lgcov
-    )
+  # Only link in the gcov library if the compiler ID is 'GNU'.
+  # Clang doesn't need a separate library for coverage.
+  if(CMAKE_C_COMPILER_ID STREQUAL "GNU")
+    target_link_libraries(testbinary PRIVATE -lgcov)
```

Comment:
@stephanosio provided you with exact code to use, so why did you decide to go for this approach ?

Reply:
ref: https://github.com/zephyrproject-rtos/zephyr/pull/52005#discussion_r1015159774

Reply:
It didn't work.

Reply:
I'll run them again to make sure

Reply:
Yes, all instances `llvm`, `zephyr`, and `host` don't produce coverage data.

Reply:
I see, I believe that currently `cmake/linker/ld/linker_flags.cmake` is never included in unit test builds. So while the linker_flags.cmake and compiler_flags.cmake do have these properties, we're not getting them here for unit tests. I can try and add the linker and compiler cmake files but I worry that this is going to make the unittest case more complex and my understanding was that it was meant to be a much smaller and faster test.

Reply:
Ugh, I just ran more tests, and it seems that it was ignoring `ZEPHYR_TOOLCHAIN_VARIANT=llvm` I'm going to change this PR to a WIP for now until I can figure this out.

Reply:
> I see, I believe that currently cmake/linker/ld/linker_flags.cmake is never included in unit test builds.

ahh, right.
I should have noticed that this was in  the `unittest.cmake` module and not our regular code.

For historic reasons this is not using Zephyr toolchain infrastructure, which is also why you see it's ignoring the `ZEPHYR_TOOLCHAIN_VARIANT` setting, and therefore also why the proposal of using the linker property doesn't work.

Sorry for the confusion.

Reply:
@tejlmand I got it working :) I'm actually really happy with this change.